data_filter_mongodb/internal/opa/opa_test.go

```diff
@@ -0,0 +1,118 @@
+package opa
+
+import (
+	"context"
+	"log"
+	"reflect"
+	"testing"
+
+	"github.com/open-policy-agent/contrib/data_filter_mongodb/internal/mongo"
+	"go.uber.org/zap"
+)
+
+func Test_opaClient_Compile(t *testing.T) {
+	type fields struct {
+		logger      *zap.Logger
+		mongoClient mongo.MongoClient
+		policyFile  string
+	}
+	type args struct {
+		ctx   context.Context
+		input map[string]interface{}
+	}
+
+	var f fields
+	var err error
+	f.logger, err = zap.NewProduction()
+	if err != nil {
+		log.Fatalf("Failed to create zap logger %v", err)
+	}
+	f.mongoClient = mongo.NewMockMongoClient(nil, "")
+	f.policyFile = "./../../example.rego"
```

Comment:
Hardcoding the policy for testing would be fine as well.

Reply:
As this is a test file am assigning the policy file. In functionality, we read policy from a file. So hardcoding the policy would create disturbance in the normal flow.